### ISSUE TITLE
KubernetesClientException when trying to display pipelines

### ISSUE THREAD

**Issue by adietish** [redhat-developer/intellij-kubernetes#116]
* depends on #104 being merged first
* this happened when testing https://github.com/redhat-developer/intellij-kubernetes/pull/104 - see https://github.com/redhat-developer/intellij-kubernetes/pull/104/files#diff-730997764121d290dbd0f7a629bb6c684a5b0e13485adfa95176fa9cd16a3b05R49

```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://api.crc.testing:6443/apis/tekton.dev/v1alpha1/namespaces/tekton-pipelines/pipelines. Message: conversion webhook for tekton.dev/v1beta1, Kind=Pipeline failed: conversion failed to version v1alpha1 for type [kind=Pipeline group=tekton.dev version=v1beta1] -  the specified field/section is not available in v1alpha1. Received status: Status(apiVersion=v1, code=500, details=null, kind=Status, message=conversion webhook for tekton.dev/v1beta1, Kind=Pipeline failed: conversion failed to version v1alpha1 for type [kind=Pipeline group=tekton.dev version=v1beta1] -  the specified field/section is not available in v1alpha1, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:589)
	at io.fabric8.kubernetes.client.dsl.internal.RawCustomResourceOperationsImpl.makeCall(RawCustomResourceOperationsImpl.java:809)
	at io.fabric8.kubernetes.client.dsl.internal.RawCustomResourceOperationsImpl.list(RawCustomResourceOperationsImpl.java:446)
	at org.jboss.tools.intellij.kubernetes.model.resource.kubernetes.custom.NamespacedCustomResourcesProvider.loadAllResources(NamespacedCustomResourcesProvider.kt:33)
	at org.jboss.tools.intellij.kubernetes.model.resource.NamespacedResourcesProvider.getAllResources(NamespacedResourcesProvider.kt:45)
	at org.jboss.tools.intellij.kubernetes.model.resource.AbstractResourcesProvider.invalidate(AbstractResourcesProvider.kt:23)
	at org.jboss.tools.intellij.kubernetes.model.resource.NamespacedResourcesProvider.setNamespace(NamespacedResourcesProvider.kt:36)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.setCurrentNamespace(ActiveContext.kt:112)
	at org.jboss.tools.intellij.kubernetes.model.context.ActiveContext.setCurrentNamespace(ActiveContext.kt:103)
	at org.jboss.tools.intellij.kubernetes.model.ResourceModel.setCurrentNamespace(ResourceModel.kt:62)
	at org.jboss.tools.intellij.kubernetes.actions.UseNamespaceAction$actionPerformed$1.run(UseNamespaceAction.kt:31)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:727)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:403)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by lstocchi** [comment]
More info, i actually have two versions of pipeline crd in my cluster 
![image](https://user-images.githubusercontent.com/49404737/98526453-0e86c580-227a-11eb-97da-5537018c0620.png)

So as you said the problem is in https://github.com/redhat-developer/intellij-kubernetes/pull/104/files#diff-730997764121d290dbd0f7a629bb6c684a5b0e13485adfa95176fa9cd16a3b05R49 as it selects the first (v1alpha1)

To replicate it you can use the pipeline below. It contains the `finally` section that doesn't exists in alpha1.
```
apiVersion: tekton.dev/v1beta1
kind: Pipeline
metadata:
  name: not-working-pp
spec:
  finally:
  - name: clear
    params:
    - name: a
      value: "1"
    - name: b
      value: "1"
    taskRef:
      kind: Task
      name: multiply-params
  - name: final
    taskRef:
      kind: Task
      name: echo-true
  params:
  - name: p1
    type: string
  - name: p2
    type: string
  tasks:
  - name: foo
    params:
    - name: res
      value: dd
    taskRef:
      kind: Task
      name: echo-string
  - name: second
    params:
    - name: a
      value: $(params.p1)
    - name: b
      value: $(params.p2)
    taskRef:
      kind: Task
      name: sum-params
```

**Comment by adietish** [comment]
The version to be displayed is the version with the highest priority. The highest priority is to be determined according to rules defined in the kubernetes docs (https://v1-17.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/):
```
Version priority

Regardless of the order in which versions are defined in a CustomResourceDefinition, the version with the highest priority is used by kubectl as the default version to access objects. The priority is determined by parsing the name field to determine the version number, the stability (GA, Beta, or Alpha), and the sequence within that stability level.

The algorithm used for sorting the versions is designed to sort versions in the same way that the Kubernetes project sorts Kubernetes versions. Versions start with a v followed by a number, an optional beta or alpha designation, and optional additional numeric versioning information. Broadly, a version string might look like v2 or v2beta1. Versions are sorted using the following algorithm:

* Entries that follow Kubernetes version patterns are sorted before those that do not.
* For entries that follow Kubernetes version patterns, the numeric portions of the version string is sorted largest to smallest.
* If the strings beta or alpha follow the first numeric portion, they sorted in that order, after the equivalent string without the beta or alpha suffix (which is presumed to be the GA version).
* If another number follows the beta, or alpha, those numbers are also sorted from largest to smallest.
* Strings that don’t fit the above format are sorted alphabetically and the numeric portions are not treated specially. Notice that in the example below, foo1 is sorted above foo10. This is different from the sorting of the numeric portion of entries that do follow the Kubernetes version patterns.
```

**Comment by adietish** [comment]
It is also completely safe to request older versions:
> Once the new version is added, clients may incrementally migrate to the new version. It is perfectly safe for some clients to use the old version while others use the new version

**Comment by adietish** [comment]
to sum it up, there are 2 things to note here:
* the exception is due to a misconfigured cluster which does not have the conversion webhook configured properly (v1alpha1 is requested, v1beta1 is stored and should be converted and delivered as v1alpha1. This fails due to a badly configured/missing webhook)
* the plugin should always request the version with the highest priority (which is defined in the kubernetes documentation quoted above)

**Comment by adietish** [comment]
I filed a separate issue where I state that we should display the custom resources in the version with the highest priority: #117

**Comment by adietish** [comment]
Closing this issue as this is obviously caused by a misconfigured cluster.
